.github/workflows/code_quality.yaml

```diff
@@ -84,15 +82,33 @@ jobs:
   ruff:
     name: ruff
     runs-on: ubuntu-latest
+    env:
+      poetry_version: '1.7.1'
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: Cache poetry in ~/.local
+        uses: actions/cache@v3
+        id: cached-home-local
+        with:
+          path: ~/.local
+          key: ${{ runner.os }}-local-${{ env.poetry_version }}
+
+      - name: Install poetry
+        if: steps.cached-home-local.outputs.cache-hit != 'true'
+        run: pip install poetry==${{ env.poetry_version }}
+
+      - name: Set up Python
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          cache: 'poetry'
+
+      - run: poetry check --lock
+
       - name: Install ruff
-        run: |
-          pip install poetry
-          poetry install --only dev
+        run: poetry install --only=dev --sync
```

Comment:
why installing once again?

Reply:
The previous install only got us the poetry executable itself. Now I use poetry to install the ods-ci dependencies

Reply:
oh right, stupid me, sorry.